### PR TITLE
Adding readme to bsp targets for E20, E21, E31/Arty, S51/Arty

### PR DESCRIPTION
**Pull request by bsousi5**

### CODE REVIEW

File: bsp/coreip-e31-arty/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+The SiFive E31 Standard Core is the world’s most deployed RISC-V core. Co-designed alongside the RISC-V ISA, the E31 takes maximum advantage of the RISC-V ISA, resulting in a power-efficient core that delivers the high performance needed for tomorrow’s smart IoT, storage, and industrial applications.
+
+This FPGA core target is ideal maker and hobbist to develop running application software building on top of freedom-metal libraries. The target supports:
```
Comment (mwachs5) [inline bsp/coreip-e31-arty/README.md]: ideal for makers and hobbiests ?

File: bsp/coreip-e31-arty/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+        - 1 hart with RV32IMAC core
+        - 4 hardware breakpoints
+        - Physical Mempory Protectin with 8 regions
```
Comment (mwachs5) [inline bsp/coreip-e31-arty/README.md]: Protection

File: bsp/coreip-e31-arty/README.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+        - 16 local interrupts signal that can be connected to off core complex devices
```
Comment (mwachs5) [inline bsp/coreip-e31-arty/README.md]: interrupt signals

File: bsp/coreip-e31-arty/README.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+        - Up to 127 PLIC interrupt signals that can be connected to off core complex devices, with 7 priority levels
+        - GPIO memory with 16 interrupt lines
+        - SPI memory with 1 intterupt line
```
Comment (mwachs5) [inline bsp/coreip-e31-arty/README.md]: interrupt... can we some how spell check all these?

File: bsp/sifive-hifive1/README.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
+HiFive1 is a low-cost, Arduino-compatible development board featuring the Freedom E310. It’s the best way to start prototyping and developing your RISC‑V applications.
+
+his FPGA core target is ideal maker and hobbist to develop running application software building on top of freedom-metal libraries. The target supports:
```
Comment (mwachs5) [inline bsp/sifive-hifive1/README.md]: This FPGA

**Comment by bsousi5** [comment]
Thanks Megan.
Will spell check all before merging. I was not sure about the content coverage and want to see how it is received.

Regards,
Bunnaroath

**Comment by sifivekevin** [comment]
Bunnaroath,  I committed formatting fixes for the bullet lists.  Removed leading white-space from list items and added blank lines before each list to separate from paragraph text.  Without these changes the markdown does not render correctly.

snag_558c6eb

**Comment by solomatnikov** [comment]
E20 does not have PMPs

**Comment by sifivekevin** [comment]
FWIW, I've updated Freedom Studio to show this when no README.md is found for bsp's and examples:

snag_5970ba8

Hopefully users will never see this.

**Pull request merged**